api/src/main/java/org/apache/gravitino/policy/Policy.java

```diff
@@ -0,0 +1,141 @@
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.gravitino.policy;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import org.apache.gravitino.Auditable;
+import org.apache.gravitino.MetadataObject;
+import org.apache.gravitino.annotation.Evolving;
+import org.apache.gravitino.exceptions.IllegalPolicyException;
+
+/**
+ * The interface of the policy. The policy is a set of rules that can be applied to a metadata
+ * object. The policy can be used for data governance and so on.
+ */
+@Evolving
+public interface Policy extends Auditable {
+
+  /**
+   * Get the name of the policy.
+   *
+   * @return The name of the policy.
+   */
+  String name();
+
+  /**
+   * Get the type of the policy.
+   *
+   * @return The type of the policy.
+   */
+  String type();
+
+  /**
+   * Get the comment of the policy.
+   *
+   * @return The comment of the policy.
+   */
+  String comment();
+
+  /**
+   * Whether the policy is enabled or not.
+   *
+   * @return True if the policy is enabled, false otherwise.
+   */
+  boolean enabled();
+
+  /**
+   * Whether the policy is exclusive or not. If the policy is exclusive, only one of the same type
+   * policy can be applied to the same object. If the policy is not exclusive, multiple policies of
+   * the same type can be applied to the same object.
+   *
+   * @return True if the policy is exclusive, false otherwise.
+   */
+  boolean exclusive();
+
+  /**
+   * Whether the policy is inheritable or not. If the policy is inheritable, it can be inherited by
+   * child objects. If the policy is not inheritable, it can only be applied to the metadata object
+   * itself.
+   *
+   * @return True if the policy is inheritable, false otherwise.
+   */
+  boolean inheritable();
+
+  /**
+   * Get the set of the metadata object types that the policy can be applied to.
+   *
+   * @return The set of the metadata object types that the policy can be applied to.
+   */
+  Set<MetadataObject.Type> supportedObjectTypes();
+
+  /**
+   * Get the content of the policy.
+   *
+   * @return The content of the policy.
+   */
+  Content content();
+
+  /**
+   * Check if the policy is inherited from a parent object or not.
+   *
+   * <p>Note: The return value is optional, Only when the policy is associated with a metadata
+   * object, and called from the metadata object, the return value will be present. Otherwise, it
+   * will be empty.
+   *
+   * @return True if the policy is inherited, false if it is owned by the object itself. Empty if
+   *     the policy is not associated with any object.
```

Comment:
"Associate" or "apply"? You should unify the terminology, the above exception is `PolicyAlreadyAppliedException`.